tests/testthat/test-rd-html.R

```diff
@@ -346,3 +346,28 @@ test_that("titles don't get autolinked code", {
   rd <- rd_text("\\title{\\code{foo()}}", fragment = FALSE)
   expect_equal(extract_title(rd), "<code>foo()</code>")
 })
+
+# Handling URL failures verbosely -----------------------------------------
+
+
+test_that("URLs handled well", {
+
+  ## Binding the rdname to the context environment allows
+  # stop_on_bad_html_tag() to grab the filename via rlang
+  rlang::env_bind(.env = context, rdname = "test-rd-html.R")
+  ## Without this, the error message in this test file is:
+  # Error: Context `rdname` has not been initialised
+
+  ## Good URLs do not throw erros and complete as desired
+  expect_equal(
+    rd2html(x = "\\url{http://google.com}"),
+    a("http://google.com", href = "http://google.com")
+  )
+
+  ## Empty URLs throw errors
+  expect_error( rd2html(x = "\\url{}") )
```

Comment:
You need to include a fragment of the error message to match on to ensure that you're finding the expected error.